libs/src/sanityChecks/needsRecoveryEmail.js

```diff
@@ -0,0 +1,13 @@
+/** Sanity check for whether a user needs a recovery email to be sent. */
+const needsRecoveryEmail = async (libs) => {
+  const user = libs.userStateManager.getCurrentUser()
+  if (!user || !user.wallet) return
+
+  const events = await libs.identityService.getUserEvents(user.wallet)
+  if (events.needsRecoveryEmail) {
```

Comment:
@raymondjacobson super edge case but if in the user events route has a invalid wallet or no entry(see comment), this might error because property doesn't exist on object

Reply:
a = {}
a.blah

doesn't throw, it's just undefined

Reply:
@raymondjacobson ah yes you are correct